src/ssh/azext_ssh/ssh_utils.py

```diff
@@ -35,43 +35,61 @@ def start_ssh_connection(op_info, delete_keys, delete_cert):
         # When connecting to a local user on a host with a banner, output gets messed up if stderr redirected.
         # If user expects logs to be printed, do not redirect logs. In some ocasions output gets messed up.
         redirect_stderr = set(['-v', '-vv', '-vvv']).isdisjoint(ssh_arg_list) and \
-            (delete_cert or op_info.delete_credentials)
-
+            (delete_cert or op_info.delete_credentials or op_info.new_service_config)
         if redirect_stderr:
             ssh_arg_list = ['-v'] + ssh_arg_list
 
         env = os.environ.copy()
         if op_info.is_arc():
             env['SSHPROXY_RELAY_INFO'] = connectivity_utils.format_relay_info_string(op_info.relay_info)
 
+        retry_attempt = 0
+        retry_attempts_allowed = 0
+        successful_connection = False
+        ssh_process = None
+        connection_duration = None
+
         # Get ssh client before starting the clean up process in case there is an error in getting client.
         command = [get_ssh_client_path('ssh', op_info.ssh_client_folder), op_info.get_host(), "-l", op_info.local_user]
 
         command = command + op_info.build_args() + ssh_arg_list
-
-        connection_duration = time.time()
         logger.debug("Running ssh command %s", ' '.join(command))
 
-        try:
-            # pylint: disable=consider-using-with
-            if redirect_stderr:
-                ssh_process = subprocess.Popen(command, stderr=subprocess.PIPE, env=env, encoding='utf-8')
-                _read_ssh_logs(ssh_process, op_info, delete_cert, delete_keys)
-            else:
-                ssh_process = subprocess.Popen(command, env=env, encoding='utf-8')
-                _wait_to_delete_credentials(ssh_process, op_info, delete_cert, delete_keys)
-        except OSError as e:
-            colorama.init()
-            raise azclierror.BadRequestError(f"Failed to run ssh command with error: {str(e)}.",
-                                             const.RECOMMENDATION_SSH_CLIENT_NOT_FOUND)
-
-        connection_duration = (time.time() - connection_duration) / 60
+        while (retry_attempt <= retry_attempts_allowed and not successful_connection):
+            service_config_delay_error_logs = False
+            if retry_attempt == 1:
+                logger.warning(f"SSH connection failed, possibly caused by new service configuration setup. "
+                               f"Retrying the connection in {str(const.RETRY_DELAY_IN_SECONDS)} seconds.")
+                time.sleep(const.RETRY_DELAY_IN_SECONDS)
+            connection_duration = time.time()
+            try:
+                # pylint: disable=consider-using-with
+                if redirect_stderr:
+                    ssh_process = subprocess.Popen(command, stderr=subprocess.PIPE, env=env, encoding='utf-8')
+                    service_config_delay_error_logs = _read_ssh_logs(ssh_process, op_info, delete_cert, delete_keys)
```

Comment:
We might want to rename this function to reflect more on what it actually does, now that it returns a new value.